### ISSUE TITLE
OM System OM-1

### ISSUE THREAD

**Issue by sarunasb** [darktable-org/darktable#11258]
Please add support for OM-1. Default 20Mpx and three hi-res mode (25, 50 and 80Mpx) RAW sampes uploaded to raw.pixl.us.

**Comment by Nilvus** [comment]
Why post 4 issues? One issue per camera is enough. And which mark you post here? Please update this issue by adding all wanted things from your 3 other issues and update title with mark you post. Thanks

**Comment by sarunasb** [comment]
> And which mark you post here? 

If mark is necessary, then it would be I. However it is not part of the model name and we don't know if there ever will be next marks of this model.

**Comment by Nilvus** [comment]
Seems that I made some confusion with E-M1 one. Sorry. didn't know that Olympus release this new model. Thanks for your update. I let people who update cameras for darktable to check your samples.

**Comment by algreen05** [comment]
I tested OM-1 support today in the latest git code and it does not work.

**Comment by Nilvus** [comment]
That's why this issue is opened @algreen05

**Comment by algreen05** [comment]
Great.  What is needed to close it?

**Comment by algreen05** [comment]
I have tested OM-1 Support in Darktable-Dev 716.  Lens Correction has the OM-1 Camera model and can recognize the camera in JPG files.  But Darktable is unable to read the OM-1 RAW .ORF files.

**Comment by Nilvus** [comment]
> I have tested OM-1 Support in Darktable-Dev 716. Lens Correction has the OM-1 Camera model and can recognize the camera in JPG files. But Darktable is unable to read the OM-1 RAW .ORF files.

So good to read that: https://www.darktable.org/2012/10/whats-involved-with-adding-support-for-new-cameras/
And remember that this is a free software made by people in their free time. So support will be added when someone will do it.

**Comment by algreen05** [comment]
Time is the only luxury we all possess and it isn't free.

**Comment by sarunasb** [comment]
Here is a branch with OM-1 support based on darktable release 4.0.0 tag.
```
git clone https://github.com/sarunasb/darktable.git
cd darktable
git checkout release-4.0.0-om1
git submodule init
git submodule update
mkdir build
cd build
cmake -DCMAKE_INSTALL_PREFIX=/opt/darktable/ ..
make -j
sudo make install
```
```
/opt/darktable/bin/darktable --version
this is darktable 4.0.om1
...
```

**Comment by sauravpandit** [comment]
Can someone guide me how I can use it in windows11. I am also not able to edit photos taken by OM-1

**Comment by luisfl** [comment]
> Here is a branch with OM-1 support based on darktable release 4.0.0 tag.
> 
> ```
> git clone https://github.com/sarunasb/darktable.git
> cd darktable
> git checkout release-4.0.0-om1
> git submodule init
> git submodule update
> mkdir build
> cd build
> cmake -DCMAKE_INSTALL_PREFIX=/opt/darktable/ ..
> make -j
> sudo make install
> ```
> 
> ```
> /opt/darktable/bin/darktable --version
> this is darktable 4.0.om1
> ...
> ```

I am playing with your custom build, and it seems to me that the highlight recovery does not work properly.
The clipping zones are replaced with blue instead of white or with the corresponding nearby matching color.
The problem is not in the highlight recovery module, but in the color calibration module.

Thanks,
L.

**Comment by sarunasb** [comment]
> I am playing with your custom build, and it seems to me that the highlight reconstruction does not work properly when "clip highlights" is selected

I don't think this is related to -om1. Highlight recovery module behaves the same in 4.0.1, with no om1 support.

**Comment by luisfl** [comment]
> @luisfl
> 
> > I am playing with your custom build, and it seems to me that the highlight reconstruction does not work properly when "clip highlights" is selected
> 
> I don't think this is related to -om1. Highlight recovery module behaves the same in 4.0.1, with no om1 support.

Indeed, you are right. Darktable highlight recovery is not working properly even in the official 4.0.1 build. In fact, none of the 4 methods work properly. Compare the result with RawTherapee:

![MyScreenshot_221104_143346](https://user-images.githubusercontent.com/15525268/200040060-34da036d-075b-45ea-a124-9e085377f727.jpg)
![MyScreenshot_221104_143537](https://user-images.githubusercontent.com/15525268/200040069-68314129-4191-48e0-acc0-fe4420a01cb7.jpg)

**Comment by luisfl** [comment]
> Here is a branch with OM-1 support based on darktable release 4.0.0 tag.
> 
> ```
> git clone https://github.com/sarunasb/darktable.git
> cd darktable
> git checkout release-4.0.0-om1
> git submodule init
> git submodule update
> mkdir build
> cd build
> cmake -DCMAKE_INSTALL_PREFIX=/opt/darktable/ ..
> make -j
> sudo make install
> ```
> 
> ```
> /opt/darktable/bin/darktable --version
> this is darktable 4.0.om1
> ...
> ```

I've been using your build with my OM-1 based on DT 4.0.
Do you plan to release a build based on the newly released 4.2?
It would be fantastic!

Thanks!!!!
L.

**Comment by sarunasb** [comment]
> Do you plan to release a build based on the newly released 4.2?
Yes, I do, maybe in a couple of days.

**Comment by esm7** [comment]
Is there any feedback somewhere from the Darktable maintainers about why this hasn't been merged yet? It's been tested by so many users by now... I've been using it for months, and thousands of photos.

**Comment by sarunasb** [comment]
rawspeed submodule has only one maintainer. He couldn't yet devote time to it… :(

**Comment by sarunasb** [comment]
> I've been using your build with my OM-1 based on DT 4.0. Do you plan to release a build based on the newly released 4.2? It would be great since it seems that the highlight recovery bugs have been addressed, at least to some extent.

I just made a branch starting with darktable 4.2 release tag. It should automatically pull the right rawspeed submodule with OM System and OM-1 support. (WB presets and noise profiles for OM-1 are added to JSON files within the main darktable tree.)

To get the source code:

```
git clone https://github.com/sarunasb/darktable.git
cd darktable
git checkout 4.2-om
git submodule init
git submodule update
```
To compile on Ubuntu I do:
```
mkdir build
cd build
cmake -DCMAKE_INSTALL_PREFIX=/opt/darktable/ ..
make -j
sudo make install
```
Running `/opt/darktable/bin/darktable --version` should give:
```
this is darktable 4.2.om1
…
```
In case your previous darktable version was earlier than 4.2, then running this will upgrade database and possibly config. One can run this newly compiled 4.2.om1 something like this to preserve the existing DB and config:
```
/opt/darktable/bin/darktable --configdir "~/.config/darktable-test"
```

**Comment by luisfl** [comment]
> > I've been using your build with my OM-1 based on DT 4.0. Do you plan to release a build based on the newly released 4.2? It would be great since it seems that the highlight recovery bugs have been addressed, at least to some extent.
> 
> I just made a branch starting with darktable 4.2 release tag. It should automatically pull the right rawspeed submodule with OM System and OM-1 support. (WB presets and noise profiles for OM-1 are added to JSON files within the main darktable tree.)
> 
> To get the source code:
> 
> ```
> git clone https://github.com/sarunasb/darktable.git
> cd darktable
> git checkout 4.2-om
> git submodule init
> git submodule update
> ```
> 
> To compile on Ubuntu I do:
> 
> ```
> mkdir build
> cd build
> cmake -DCMAKE_INSTALL_PREFIX=/opt/darktable/ ..
> make -j
> sudo make install
> ```
> 
> Running `/opt/darktable/bin/darktable --version` should give:
> 
> ```
> this is darktable 4.2.om1
> …
> ```
> 
> In case your previous darktable version was earlier than 4.2, then running this will upgrade database and possibly config. One can run this newly compiled 4.2.om1 something like this to preserve the existing DB and config:
> 
> ```
> /opt/darktable/bin/darktable --configdir "~/.config/darktable-test"
> ```

I just compiled and installed your new 4.2 version in Fedora 36 Šarūnas, and it seems to be working great.

Nice 2023 and THANKS A LOT!!!!!!!!!!!!!

L.

**Comment by dadu007** [comment]
Could some kind soul dumb it down for this lowly Windows user? I see someone here has support for the new OM-1 working on the Linux version of Darktable. Is there an expected date (or current way) to add support for that camera in the Windows version of Darktable? Github is a strange place to me; I previously created an account to get builds of software to run a closed-loop insulin pump for my son. (I had very detailed instructions on another site that only showed how to grab the build from Github.)  All that to say, be kind. :-)  Newb to Nth newbieness.

**Comment by lintujuh** [comment]
There is a ready-build Windows package available, check the post in pixls.us https://discuss.pixls.us/t/om-1-support-in-darktable/30980/55

**Comment by lintujuh** [comment]
Hi, and thank you for your quick fixes.

Copy-Paste woks now, the only minor issue is, that when I paste, the
coordinates (lat, long, alt) are not selected in the paste dialog. I have
to manually select them.

Ignoring the reverse geolocation works also fine, tested that also. I
created also an account in GeoNames, but I got two error dialogs: first
unable to reach API, and second stating error code Unauthorized. Does the
GeoNames require some time for activating the API for a new account,
because I ran the import just after creating the account.

-Juha

**Comment by lintujuh** [comment]
Sorry for the spam, I answered to a wrong email.

-Juha

**Comment by dadu007** [comment]
> There is a ready-build Windows package available, check the post in pixls.us https://discuss.pixls.us/t/om-1-support-in-darktable/30980/55
Thanks so much! I never would have found that. I downloaded and it's working my OM-1 files on Windows 10. Much appreciated!

**Comment by Nilvus** [comment]
I close. Rawspeed support added. Remains noise profiles (another issue opened) and PR opened and to update for WB.

**Comment by TurboGit** [comment]
Just a little note, this has been added in rawspeed but the sub-module has not yet been updated so at the moment it is still not supported on darktable.

**Comment by kmilos** [comment]
Would be great to merge this into rawspeed stable and backport for 4.2.1 😉 @LebedevRI @TurboGit

**Comment by Nilvus** [comment]
Hmm, forget that step. This will be done quickly I think so we can let this one opened. Will try to remember that next time!

**Comment by lintujuh** [comment]
I have build dt from the latest master, and I'm not having OM-1 support. I have rebuilt everything from scratch, without success. Have I done something wrong, or should I wait for the submodule to be updated?

**Comment by Nilvus** [comment]
@lintujuh: you have the answer just 3 comments above.

**Comment by sarunasb** [comment]
```
cd src/external/rawspeed
git checkout develop
```
This will have OM-1. Not that I can explain it :)

**Comment by kanwarplaha** [comment]
The path `.../darktable/src/external/rawspeed` shows as failed. Also, https://github.com/sarunasb/rawspeed.git returns 404.

**Comment by sarunasb** [comment]
> So, i tried commenting out `rawspeed` from `.gitmodules`. 


> Does the `4.2-om` or `4.2.om1` not build anymore?

It should still build, those there is no need to use my branch of rawspeed anymore, as OM support has been merged into darktable-org/rawspeed.

**Comment by kanwarplaha** [comment]
I didn't pull `rawspeed` branch (or submodule) but follow these steps:

```
$ git clone https://github.com/sarunasb/darktable.git
Cloning into 'darktable'...
remote: Enumerating objects: 257102, done.
remote: Counting objects: 100% (11/11), done.
remote: Compressing objects: 100% (6/6), done.
remote: Total 257102 (delta 5), reused 10 (delta 5), pack-reused 257091
Receiving objects: 100% (257102/257102), 1.13 GiB | 3.20 MiB/s, done.
Resolving deltas: 100% (203472/203472), done.
$ cd darktable
$ git checkout 4.2-om
branch '4.2-om' set up to track 'origin/4.2-om'.
Switched to a new branch '4.2-om'
$ git submodule init
Submodule 'src/external/LibRaw' (https://github.com/LibRaw/LibRaw.git) registered for path 'src/external/LibRaw'
Submodule 'src/external/OpenCL' (https://github.com/KhronosGroup/OpenCL-Headers.git) registered for path 'src/external/OpenCL'
Submodule 'src/external/libxcf' (https://github.com/houz/libxcf.git) registered for path 'src/external/libxcf'
Submodule 'src/external/rawspeed' (https://github.com/darktable-org/rawspeed.git) registered for path 'src/external/rawspeed'
Submodule 'src/external/whereami' (https://github.com/gpakosz/whereami) registered for path 'src/external/whereami'
Submodule 'src/tests/integration' (https://github.com/darktable-org/darktable-tests.git) registered for path 'src/tests/integration'
```

**Comment by kmilos** [comment]
You can just check out master now.

**Comment by sarunasb** [comment]
Full support for OM has been merged into  [darktable main](https://github.com/darktable-org/darktable). One can just follow build instructions there.

**Comment by LebedevRI** [comment]
@sarunasb Hello. RPU is finally up. Please contribute the full sample set.

**Comment by sarunasb** [comment]
I don't have OM-1 anymore, but it looks like there are already samples from it on https://raw.pixls.us.

**Comment by LebedevRI** [comment]
Yes, i think i posted this on the wrong issue. This was certainly meant for the one about OM-3.
